extensions/amp-auto-ads/0.1/ad-network-config.js

```diff
@@ -66,6 +68,13 @@ class AdNetworkConfigDef {
    * @return {!SizeInfoDef}
    */
   getSizing() {}
+
+  /**
+   * Returns the sticky ad attributes that should be applied on sticky ad.
+   * @param {!JsonObject} unusedConfigObj
+   * @return {JsonObject<string, string>} null if there is no sticky ad attributes.
```

Comment:
Why do you need all of this here?  Instead can you put attribute extraction for sticky ads solely within AnchorAdStrategy?  All you need is the config which is accessible there.

Reply:
'no-fill' attribute is specifically related to Adsense network so to keep it under Adsense network config I moved this logic here.

Reply:
If we want this logic to live client-side, then I think what @aishwaryavora is doing here is the right approach. We are introducing the concept of a non-filled sticky ad request and also a data-no-fill attribute. Since this (at the moment) is an AdSense only concepts, it makes sense to keep the logic for it in the AdSenseNetworkConfig and not pollute the generalized amp-auto-ads code with isAdsense checks.

Another approach that was discussed, would be to just add a new field in the config that contains attributes that are specific to the anchor ad strategy. We would then not require the new OptInStatus or any of this logic client-side. All the decision logic about whether to enable anchor and whether a data-no-fill attribute should be used could then live in the ad server.